<robot xmlns:xacro="http://www.ros.org/wiki/xacro"
xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#slider"
xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">

  <xacro:include filename="$(find sr_description)/hand/xacro/thumb/thproximal.gazebo.xacro" />
  <xacro:include filename="$(find sr_description)/hand/xacro/thumb/thproximal.transmission.xacro" />
  <!-- muscle is true for muscle hand or false for motor hand -->
  <!-- reflect is either 1 (for right hand) or -1 (for left hand) -->
  <!-- prefix can be anything but usually is "r_" or  "l_" for right and left hands distinction -->
  <!-- parent is the parent link for the first joint -->
  <xacro:macro name="thproximal"
  params="muscle reflect prefix parent">
    <link name="${prefix}thproximal">
      <inertial>
        <mass value="0.040" />
        <origin xyz="0 0 0.019" rpy="0 0 0" />
        <inertia ixx="0.0000136" ixy="0.0" ixz="0.0"
        iyy="0.0000136" iyz="0.0" izz="0.00000313" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry name="${prefix}thproximal_visual">
          <mesh filename="package://sr_description/meshes/hand/TH3_z.dae"
          scale="0.001 0.001 0.001" />
        </geometry>
        <material name="shadow_thproximal_material">
          <color rgba="0.2 0.2 0.2 1.0" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0.020" rpy="0 0 0" />
        <geometry name="${prefix}thproximal_collision_geom">
          <cylinder radius="0.012" length="0.018" />
        </geometry>
      </collision>
    </link>
    <joint name="${prefix}THJ4" type="revolute">
      <parent link="${parent}" />
      <child link="${prefix}thproximal" />
      <origin xyz="0 0 0" rpy="0 0 0" />
      <axis xyz="${reflect} 0 0" />
      <limit lower="0.0" upper="${70/180*M_PI}" effort="3"
      velocity="4.0" />
      <dynamics damping="0.2" />
    </joint>
    <gazebo reference="${prefix}THJ4">
      <provideFeedback>1</provideFeedback>
      <implicitSpringDamper>1</implicitSpringDamper>
    </gazebo>
    <!-- extensions -->
    <xacro:thproximal_transmission muscletrans="${muscle}"
    prefix="${prefix}" />
    <xacro:thproximal_gazebo prefix="${prefix}" />
  </xacro:macro>
</robot>
